homeassistant/components/bryant_evolution/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for Bryant Evolution integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from evolutionhttp import BryantEvolutionClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CONF_SYSTEM_ID, CONF_ZONE_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_SYSTEM_ID, default=1): vol.All(int, vol.Range(min=1)),
+        vol.Required(CONF_ZONE_ID, default=1): vol.All(int, vol.Range(min=1)),
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    # Verify that we can read S1Z1 to check that the host is valid.
+    client = BryantEvolutionClient(data[CONF_HOST], 1, 1)
+    if await client.read_hvac_mode() is None:
+        raise CannotConnect
+    return {"title": f"System {data[CONF_SYSTEM_ID]} Zone {data[CONF_ZONE_ID]}"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Bryant Evolution."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
```

Comment:
since you're only using this once, move it inline, so you also can get rid of the exceptions

Reply:
if I do not catch the exceptions, then the config flow tests will fail -- I had taken them from the skeleton generation tool IIRC. Thoughts?

Reply:
sorry, I meant do catch them, but if you move them inline, there's no need to define custom exceptions, but you can catch them directly

Reply:
sorry, I think I am still not following. Mind sketching the code you'd like?

Reply:
what I mean is, if you put the content of `validate_input` in here, except of raising and catching `CannotConnect` you can write directly to the errors dict.

Reply:
Ah! simplified to remove exceptions; I kept a separate function to improve readability.